onnxruntime/test/shared_lib/utils.cc

```diff
@@ -2,11 +2,19 @@
 // Licensed under the MIT License.
 
 #include "utils.h"
+#include <limits>
 
 OrtCUDAProviderOptions CreateDefaultOrtCudaProviderOptionsWithCustomStream(void* cuda_compute_stream) {
   OrtCUDAProviderOptions cuda_options;
+
+  cuda_options.device_id = 0;
+  cuda_options.cudnn_conv_algo_search = OrtCudnnConvAlgoSearch::OrtCudnnConvAlgoSearchExhaustive;
+  cuda_options.gpu_mem_limit = std::numeric_limits<size_t>::max();
```

Comment:
>std::numeric_limits<size_t>::max() [](http://example.com/codeflow?start=31&length=34)

Is this a special value or GPU can indeed address this much?

Reply:
Not a special value as such and this doesn't have anything to do with the GPU (the prefix "gpu" is in there because it is part of the provider struct and for legacy reasons). This is a BFCArena knob and we are basically telling it there is "no enforced limit" to the memory size.  Of course, it is still bound by the actual memory available on the device.